core/src/com/unciv/logic/map/TileInfo.kt

```diff
@@ -319,7 +317,6 @@ open class TileInfo {
             tileUniques += city.getMatchingUniques(UniqueType.StatsFromObject, stateForConditionals)
             for (unique in localUniqueCache.get("StatsFromTilesAndObjects", tileUniques)) {
                 val tileType = unique.params[1]
-                if (tileType == improvement) continue // This is added to the calculation in getImprovementStats. we don't want to add it twice
```

Comment:
Not sure I understand why it has been removed. According to the comment, will they be counted twice now?

Reply:
The line below calls `matchesTerrainFilter`, which specifically excludes improvements from matching the unique. This line has since become unnecessary, so I removed it. I'm fine with adding it back 'just to be safe', though

Reply:
I would keep it in the name of extra-tiny-minor optimization: I assume, it is a bit faster than `matchesTerrainFilter` and at least, you will not need to test for the "Natural Wonder" later.

Reply:
Another option is to move a check for the natural wonder _before_ the terrain filter check and add `continue` in there since the wonder overrides its terrain properties (or not?).

Reply:
> since the wonder overrides its terrain properties

Not necessarily, only it `overrideStats` is set to true. I'll see if I can simplify the function a bit

Reply:
I'll see whether I can reorder things for possible efficiency